### ISSUE TITLE
Remove usage of tuple calls for OTP 21.x compatibility.

### ISSUE THREAD

**Issue by eiri** [apache/couchdb#1890]
## Summary

In OTP 21 support for "tuple calls" have been removed from run-time system (OTP-14497) and a compiler option `tuple_calls` been added to support back compatibility.

## Desired Behaviour

We should change all the instances of tuple call usage to normal `apply/3` and remove `tuple_calls` compiler flags.

## Possible Solution

The code change would look similar to this:

from
```erlang
RawUri = MochiReq:get(raw_path),
```

to
```erlang
MochiReqMod = element(1, MochiReq),
RawUri = erlang:apply(MochiReqMod, get, [raw_path, MochiReq]),
```

**Comment by wohali** [comment]
@eiri Our minimum Erlang version is 19 now; can you confirm this new syntax is supported all the way back to 19? It looks like it is, but I just want to be sure. Thanks.

**Comment by eiri** [comment]
@wohali yes, `element/2` and `apply/3` are going back as far as OTP R10 at lease and the change snippet itself is pretty much what compiler does for the tuple calls behind the scene when flag `tuple_calls` is set. The flag itself was added in OTP 21 for back compatibility and @nickva set it in #1640. We have until OTP 24 until this back compatibility removed.

There are an option not to use somehow obscure `apply` dance, but just call `mochiweb_request` directly, i.e. instead of `RawUri = MochiReq:get(raw_path),` do `RawUri = mochiweb_request:get(raw_path, MochiReq),`, but it depends on how mochiweb is going to address removing tuple calls, they may introduce now module as a frontend or start requiring `parse_transform` in style of `lager`.

**Comment by nickva** [comment]
`mochiweb_request:get(raw_path, MochiReq)` might end up being the official API. But I think we should wait for mochiweb to update it's API. Which, if it decides on the above format, would just mean switching all its examples and docs (if it has any) to say from now on the official API is `mochiweb_request:<function>(...., MochiReq)` and call it a 3.x release.

But I suspect that's at least a few years away.
